src/Select/Select.jsx

```diff
@@ -44,6 +44,22 @@ export function Select({
 }) {
   const {cssClass} = Select;
 
+  if (!lazy) {
```

Comment:
Alternatively, you can rewrite propTypes to behave a little differently to achieve the same thing without explicit prop checking code, like so:

```
Select.propTypes = {
  // ...
  options: React.PropTypes.oneOfType([
    React.PropTypes.arrayOf(selectValuePropType),
    React.PropTypes.shape({
      // custom validator to ensure that lazy must be set to true for `loadOptions` to be required
      lazy: (v) => v === true ? null : new Error("`lazy` must be `true` to use lazy loading"),
      loadOptions: React.PropTypes.func.isRequired,
    }),
  ]).isRequired,
  // ...
}
```

This way existing usage of options directly as an array is supported, but if it's an object of the lazy format it would use that form.